staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper_test.go

```diff
@@ -373,11 +463,16 @@ func TestRoundTripAndNewConnection(t *testing.T) {
 					encodedCredentials := base64.StdEncoding.EncodeToString([]byte(testCase.proxyAuth.String()))
 					expectedProxyAuth = "Basic " + encodedCredentials
 				}
-				if len(expectedProxyAuth) == 0 && proxyCalledWithAuth {
-					t.Fatalf("%s: Proxy authorization unexpected, got %q", k, proxyCalledWithAuthHeader)
+				if testCase.socks5ProxyServerFunc == nil {
+					if len(expectedProxyAuth) == 0 && proxyCalledWithAuth {
+						t.Fatalf("%s: Proxy authorization unexpected, got %q", k, proxyCalledWithAuthHeader)
+					}
+					if proxyCalledWithAuthHeader != expectedProxyAuth {
+						t.Fatalf("%s: Expected to see a call to the proxy with credentials %q, got %q", k, testCase.proxyAuth, proxyCalledWithAuthHeader)
+					}
```

Comment:
this change is sort of hard to follow... I would have thought the expectedProxyAuth/proxyCalledWithAuth and `proxyCalledWithAuthHeader != expectedProxyAuth` checks could have remained as-is if the socks5 test proxy server recorded the fact that it was called with auth and the specific auth it was called with similarly